dev/com.ibm.ws.jaxrs.2.0.server/src/com/ibm/ws/jaxrs20/server/LibertyJaxRsServerFactoryBean.java

```diff
@@ -283,7 +291,13 @@ protected synchronized void doInit() throws ServletException {
 
         //step 3: prepare the EJB or CDI provider & resources to EJB or CDI's beanCustomizer
         for (JaxRsFactoryBeanCustomizer customizer : beanCustomizers) {
+            String key = createCustomizerKey(customizer);
+            Object oldCustomizerContext = beanCustomizerContexts.get(key);
             BeanCustomizerContext context = new BeanCustomizerContext(endpointInfo, moduleMetadata, cxfPRHolder);
+
+            if (key != null && (oldCustomizerContext instanceof HashMap)) {
```

Comment:
This might be better if it checked for an instance of `Map` as opposed to `HashMap` in case we decide to use different types of Maps in the future - I'm not sure, but I think we currently only use HashMaps (and subclasses of HashMaps - like ThreadBasedHashMap), so this is probably safe as-is, but would be safer to change to check for `Map`.

Reply:
I chose HashMap in order to use the Map we were using... but I see your point for future flexibility.   I'll change this.